### DISCUSSION TITLE
Create a texture: Stride is too small.

### DISCUSSION THREAD

**Discussion by FlightBlaze** [DiligentGraphics/DiligentEngine#170] [Q&A] [answered]
Hi! When I call `renderDevice->createTexture()`, Diligent is asserting me:
```
Diligent Engine: ERROR: Debug assertion failed in InitializeTextureContent(), file TextureVkImpl.cpp, line 317:
Stride is too small
```
If I comment the assertions in TextureVkImpl.cpp, everything will work just fine.

**About the image:**

```
604 x 604px
3 channels (RGB)
channel depth = 8
```

**About the texture:**

```
Size is the same
Format: TEX_FORMAT_RGBA8_UNORM_SRGB
Mip levels: 1
```

**How stride is calculated:**

stride = width * channels

**What stride size is expected by Diligent:**

width * 4

**Comment by TheMostDiligent** [comment, marked as answer]
> What stride size is expected by Diligent:
> 
> width * 4

Yes, you are creating a four-channel RGBA texture, so Diligent expects 4-channel input data. There are no 3-channel textures in graphics APIs.

**Reply by FlightBlaze** [reply]
Thanks! I converted the RGB image to RGBA and Diligent no longer swears!

**Comment by TheMostDiligent** [comment]
FYI, there are texture loading utilities: https://github.com/DiligentGraphics/DiligentTools/blob/master/TextureLoader/interface/TextureLoader.h
